runelite-client/src/main/java/net/runelite/client/plugins/timersandbuffs/TimersAndBuffsPlugin.java

```diff
@@ -479,13 +480,19 @@ else if (nextSuperAntifireTick - tickCount <= 0)
 				|| client.getVarbitValue(Varbits.DIVINE_BATTLEMAGE) > event.getValue()
 				// When drinking a dose of moonlight potion while already under its effects, desync between
 				// Varbits.MOONLIGHT_POTION and Varbits.DIVINE_SUPER_DEFENCE can occur, with the latter being 1 tick
-				// greater
-				|| client.getVarbitValue(Varbits.MOONLIGHT_POTION) >= event.getValue())
+				// greater due to Varbits.MOONLIGHT_POTION ticking down twice in its first tick.
+
+				// When the user drinks a dose of moonlight potion at the exact tick the potion runs out, i.e. right
+				// after Varbits.MOONLIGHT_POTION == 0, the first tick behaves exactly as the above described desync.
+				// However, after the first tick, Varbits.MOONLIGHT_POTION is suddenly transmitted BEFORE
+				// Varbits.DIVINE_SUPER_DEFENCE every tick. Thus, when the latter is transmitted, the
+				// Varbits.MOONLIGHT_POTION value might be 1 less than the Varbits.DIVINE_SUPER_DEFENCE value.
+				|| client.getVarbitValue(Varbits.MOONLIGHT_POTION) + 1 >= event.getValue())
 			{
 				return;
 			}
 
-			if (client.getVarbitValue(Varbits.MOONLIGHT_POTION) < event.getValue())
+			if (client.getVarbitValue(Varbits.MOONLIGHT_POTION) + 1 < event.getValue())
 			{
 				removeVarTimer(MOONLIGHT_POTION);
 			}
```

Comment:
While slightly unrelated to the changes at hand in this PR, I'm trying to remember why I recommended adding this before. What actually happens when a divine super defence potion is drank before a moonlight potion ends? If I recall correctly, it causes the defense boost to drain normally for the duration of the divine effect (down to normal boost levels, I assume), then removes the boost at the end of it.

Would you mind confirming that behavior? I'll want to either remove this (if both timers should be preserved) or add some code to the moonlight timer branch to avoid re-adding it on further var updates.

Reply:
Your original quote is https://github.com/runelite/runelite/pull/17662#discussion_r1677034740
> Also, similarly to how drinking a moonlight potion removes existing divine super defence potion infoboxes, drinking a divine super defence potion (or having this var update from any of the other divine potions which provide its effect) should remove the moonlight potion's infobox, since it preserves the defence boost beyond the 5 minutes provided by the moonlight potion.

To be frank, it had been a couple months since I had worked on this when you requested the change (I probably didn't have all the very specific interactions still in my head), it was a quite direct request (i.e. not a question), and it didn't seem like it'd e.g. break anything significantly, so I probably didn't review your requested changes as critically and test them as extensively as I should have at the time. My apologies.


The behaviour you currently describe is indeed what I seem to remember, and it's also what I wrote down in one of the posts of my previous PR, I think. I re-tested it:
Drinking a divine super def after a moonlight potion using this feature branch:
- The defence buff ticks down to divine super def levels (118) and gets removed when the divine super def ends. Video of test: https://youtu.be/qiPKAjWhjvw
  - It really ticks down to 118. It's not like the def level gets set to 118 when the moonlight timer ends. In the video, you can see my def is boosted to 119 while the moonlight effect has already worn off.
- The moonlight infobox is currently not properly removed when the super divine def is consumed after the moonlight potion, since the moonlight timer is immediately being readded (the early return when ``Varbits.MOONLIGHT_POTION`` updates is missing). This could be seen in the previous video, alternatively with varbits: https://youtu.be/oXqQERuRMcM


I don't really mind either...
- showing both the moonlight timer and divine super defence (removing the if statement you mentioned), or...
- actually removing the moonlight timer by adding an early return when ``Varbits.MOONLIGHT_POTION`` updates, and adding the test you wrote.

For now I've pushed a commit that does the former. Feel free to let me know if you prefer the latter instead!
I'll do more extensive re-testing to confirm everything is still working as expected soon(tm).